Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -827,18 +827,41 @@ public final class ManifestLoader: ManifestLoaderProtocol {
                         // This provides some safety against arbitrary code execution when parsing manifest files.
                         // We only allow the permissions which are absolutely necessary.
                         if self.isManifestSandboxEnabled {
-                            let cacheDirectories = [self.databaseCacheDir, moduleCachePath].compactMap{ $0 }
-                            let strictness: Sandbox.Strictness = toolsVersion < .v5_3 ? .manifest_pre_53 : .default
-                            cmd = Sandbox.apply(command: cmd, strictness: strictness, writableDirectories: cacheDirectories)
+                            var sandbox = SandboxProfile()
+                            
+                            // Allow writing inside the temporary directory.
+                            sandbox.pathAccessRules.append(.writable(localFileSystem.tempDirectory))
+
+                            // Allow writing in the database cache directory, if we have one.
+                            if let databaseCacheDir = self.databaseCacheDir {
+                                sandbox.pathAccessRules.append(.writable(databaseCacheDir))
+                            }
+
+                            // Allow writing in the module cache path, if there is one.
+                            if let moduleCachePath = moduleCachePath {
+                                sandbox.pathAccessRules.append(.writable(moduleCachePath))
+                            }
+
+                            // But do not allow writing in the directory that contains the manifest, even if it is
+                            // inside one of the otherwise writable directories.
+                            sandbox.pathAccessRules.append(.readonly(manifestPath.parentDirectory))
+
+                            // Finally apply the sandbox.
+                            cmd = sandbox.apply(to: cmd)
                         }
 
-                        // Run the compiled manifest.
+                        // Set up the environment so that the manifest inherits our own environment, but make sure that
+                        // `TMPDIR` is set to whatever the temporary directory is that is allowed in the sandbox.
                         var environment = ProcessEnv.vars
+                        environment["TMPDIR"] = localFileSystem.tempDirectory.pathString
```

Comment:
hard coding to localFileSystem okay in this case?

Reply:
Yes, because this needs to match the sandbox (and when the manifest does file system operations, it's always in the context of the local file system).